### PR TITLE
Create typegen-test.yml

### PR DESCRIPTION
**Pull request by AloeareV**
Add CI to run `cargo build`, `cargo test`, and `cargo run -- generate` for `zcashrpc-typegen`. Will not test `zcashrpc` or `zcash-rcli` as they depend on running instance of regtest zcashd, which will prove more difficult to set up.

### CODE REVIEW

**Comment by AloeareV** [comment]
Well, this needs more work. Uh. I think I'm about done for tonight, hopefully by next time I can at least get `cd` working

**Comment by dannasessha** [comment]
Good we're getting tests going!

Bumping.

**Comment by dannasessha** [comment]
Once landed, this should solve issue #29.

**Comment by AloeareV** [comment]
So: The tests fail.. Because of the error we already know about, the one that we can fix by getting rid of the `*` in favor of explicitly using tokio version `0.2.22`. I vote we merge this, afaict it's set up to run the tests against a running regtest zcashd instance.

File: .github/workflows/zcashrpc-test.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: cargo-test-zcashrpc-typegen
+
+on:
+  pull_request:
+    branches: [ master ]
+    types: [ assigned, opened, synchronize, reopened ]
```
Comment (zancas) [inline .github/workflows/zcashrpc-test.yml:6]: I haven't drilled all the way down yet, but from a cursory inspection of this:

https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/github-event-types#pullrequestevent

I wonder if we should trigger on "assigned" and "synchronized"?

File: .github/workflows/zcashrpc-test.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name:  Authenticate Github CR
```
Comment (zancas) [inline .github/workflows/zcashrpc-test.yml:15]: "CR"?

File: .github/workflows/zcashrpc-test.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      env: 
+        PAT: ${{ secrets.CONTAINER_REGESTRY_READ }}
```
Comment (zancas) [inline .github/workflows/zcashrpc-test.yml:17]: Typo:   "REGESTRY" --> "REGISTRY"?

File: .github/workflows/zcashrpc-test.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      run: echo $PAT | docker login ghcr.io -u USERNAME --password-stdin
```
Comment (zancas) [inline .github/workflows/zcashrpc-test.yml:18]: Can we switch to token-based or better `ssh`-based login?

**Review by zancas — CHANGES_REQUESTED** [review]

File: .github/workflows/zcashrpc-test.yml (hunk 1 above)
Comment (AloeareV) [reply .github/workflows/zcashrpc-test.yml:6]: I'm.. Not entirely sure. I'll look into this further.

File: .github/workflows/zcashrpc-test.yml (hunk 2 above)
Comment (AloeareV) [reply .github/workflows/zcashrpc-test.yml:15]: Container Registry

File: .github/workflows/zcashrpc-test.yml (hunk 3 above)
Comment (AloeareV) [reply .github/workflows/zcashrpc-test.yml:17]: Ah. Yes. I think I can fix this easily enough

File: .github/workflows/zcashrpc-test.yml (hunk 4 above)
Comment (AloeareV) [reply .github/workflows/zcashrpc-test.yml:18]: The github container registry is in open beta, and as of yet only supports personal access token and password based login.

File: .github/workflows/zcashrpc-test.yml (hunk 3 above)
Comment (zancas) [reply .github/workflows/zcashrpc-test.yml:17]: Can you resolve this conversation, or do I have to?   Or is it configurable?

File: .github/workflows/zcashrpc-test.yml (hunk 4 above)
Comment (zancas) [reply .github/workflows/zcashrpc-test.yml:18]: What are the tradeoffs between Personal Access Tokens and Passwords?   I have a notion that PATs are more secure but I'd like to discuss it.

File: .github/workflows/zcashrpc-test.yml (hunk 3 above)
Comment (AloeareV) [reply .github/workflows/zcashrpc-test.yml:17]: I can, I will once I fix it!

File: .github/workflows/zcashrpc-test.yml (hunk 4 above)
Comment (AloeareV) [reply .github/workflows/zcashrpc-test.yml:18]: The PAT is limited to read only access to my container registry, whereas a password would have full account access.

**Review by zancas — APPROVED** [review]

**Pull request merged**